_notebooks/2020-07-28-lineage.ipynb

```diff
@@ -0,0 +1,323 @@
+{
+ "cells": [
+   {
+    "cell_type": "markdown",
+    "metadata": {},
+    "source": [
+     "# Record metadata on Kubeflow from Notebooks\n",
+     "> Demonstration of how lineage tracking works\n",
+     "\n",
+     "- toc: true \n",
+     "- badges: true\n",
+     "- comments: true\n",
+     "- categories: [jupyter]"
+    ]
+   },
+   {
+    "cell_type": "markdown",
+    "metadata": {},
+    "source": [
+     "# Lineage Tracking\n",
+     "* The intent of this blog is to highlight how people can use metadata to do lineage tracking.\n",
+     "* We are going to create notebook that uses the metadata SDK to write a sequence of executions and artifacts to TFX metadata.\n",
+     "* We are going to open up lineage view in the KFP UI to view the lineage\n",
+     "* Here shows explanation of functionality of metadata sdk, but if you want to try and deploy model and want to see lineage tracking UI then skip to plot lineage UI section [Plot the lineage graph](http://0.0.0.0:4000/jupyter/2020/07/28/lineage.html#Plot-the-lineage-graph)"
+    ]
+   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Install the _Kubeflow-metadata_ library"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# To use the latest publish `kubeflow-metadata` library, you can run:\n",
+    "!pip install kubeflow-metadata --user\n",
+    "# Install other packages:\n",
+    "!pip install pandas --user\n",
+    "# Then restart the Notebook kernel."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas\n",
+    "from kubeflow.metadata import metadata\n",
+    "from datetime import datetime\n",
+    "from uuid import uuid4"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "METADATA_STORE_HOST = \"metadata-grpc-service.kubeflow\" # default DNS of Kubeflow Metadata gRPC serivce.\n",
+    "METADATA_STORE_PORT = 8080"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a workspace and a run in a workspace\n",
+    "* [Workspace](https://github.com/kubeflow/metadata/blob/25b44da29213968a2c438d24aad3656cc86d0499/sdk/python/kubeflow/metadata/metadata.py#L92) groups a set of runs of pipelines, notebooks and their related artifacts and executions\n",
+    "* [Store](https://github.com/kubeflow/metadata/blob/25b44da29213968a2c438d24aad3656cc86d0499/sdk/python/kubeflow/metadata/metadata.py#L59) that connects to the Metadata gRPC service\n",
```

Comment:
[Store](...)  is an object that provides a connection to the Metadata gRPC service